workflow/common/util.go

```diff
@@ -320,10 +393,12 @@ func substituteParams(tmpl *wfv1.Template, globalParams, localParams map[string]
 	// Now replace the rest of substitutions (the ones that can be made) in the template
 	replaceMap = make(map[string]string)
 	for _, inParam := range globalReplacedTmpl.Inputs.Parameters {
-		if inParam.Value == nil {
-			return nil, errors.InternalErrorf("inputs.parameters.%s had no value", inParam.Name)
+		if inParam.PassthroughRegexp == "" { //this can still be set if we're handling a WorkflowTemplate
```

Comment:
Should it be `Regexp`?

Reply:
Good catch. I've gone through all the others and think they're all correct.